src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,35 @@
-import React from 'react';
+import {
+  useEffect, useState, memo, FC,
+} from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+  onClose: () => void;
+}
+
+export const TodoModal: FC<Props> = memo(({ todo, onClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(
+    () => {
+      getUser(todo.userId)
+        .then(setUser)
+        .catch(error => {
+          throw new Error(error.statusSelect);
+        });
+    },
+    [],
```

Comment:
Add 'todo.userId' to the dependency array to make sure the user data is re-fetched when the todo userId changes.